packages/extensions-sdk/src/cli/commands/add.ts

```diff
@@ -133,6 +133,7 @@ export default async function add(): Promise<void> {
 			devDependencies: await getExtensionDevDeps(
 				newEntries.map((entry) => entry.type),
 				getLanguageFromEntries(newEntries),
+				{ base: extensionManifest.devDependencies },
 			),
```

Comment:
Honstely, I'd choose this route. Makes it easier to see what's happening to the existing deps.

```suggestion
			devDependencies: {
				...extensionManifest.devDependencies,
				...(await getExtensionDevDeps(
					newEntries.map((entry) => entry.type),
					getLanguageFromEntries(newEntries),
				)),
			},
```

Reply:
... and we don't have to care about the parameter / option name 😃

Reply:
I was originally going to go this way but decided to use the options route for the following reasons:
- slightly more performant as we are just adding keys, no double spread.
- allows us to re-use this logic without needing to spread every time we want to keep the base deps.

I do agree it might not be as clear as a spread but I was hoping the reasons above outweighed that.

Reply:
> - slightly more performant as we are just adding keys, no double spread.

In my opinion, this can be neglected in this case. If, for example, we were dealing with an operation in the API, then yes, absolutely! 🚀🙏 But for this one-off command, it has no noticeable effect.

> - allows us to re-use this logic without needing to spread every time we want to keep the base deps.

I think the function `getExtensionDevDeps` currently does exactly what it should: returning the dev dependencies that are specific to the extension.
To me it feels a bit strange to pass the existing dependencies to this function and expect it to merge them. Which is why I would prefer to do it outside of the function. 

But that probably goes in the direction of personal preference, so you have my approval and I'll leave it up to you to make the final decision.

Reply:
Fair point, I have updated the code accordingly.